queries/scala/highlights.scm

```diff
@@ -1,55 +1,62 @@
 ; CREDITS @stumash (stuart.mashaal@gmail.com)
 
+(class_definition
+  name: (identifier) @type.definition)
+
+(enum_definition
+  name: (identifier) @type.definition)
+
+(object_definition
+  name: (identifier) @type.definition)
+
+(trait_definition
+  name: (identifier) @type.definition)
+
+(full_enum_case
+  name: (identifier) @type.definition)
+
+(simple_enum_case
+  name: (identifier) @type.definition)
+
 ;; variables
 
-(identifier) @variable
+(class_parameter 
+  name: (identifier) @parameter)
 
-((identifier) @variable.builtin
- (#lua-match? @variable.builtin "^this$"))
 
 (interpolation) @none
 
-; Assume other uppercase names constants.
-; NOTE: In order to distinguish constants we highlight
-; all the identifiers that are uppercased. But this solution
-; is not suitable for all occurrences e.g. it will highlight
-; an uppercased method as a constant if used with no params.
-; Introducing highlighting for those specific cases, is probably
-; best way to resolve the issue.
-((identifier) @constant (#lua-match? @constant "^[A-Z]"))
-
 ;; types
 
+(type_definition
+  name: (type_identifier) @type.definition)
```

Comment:
is this a type alias?

Reply:
It is, yes, like

```
type MyStuff = String
```

So considering your previous comment, closer to C's typedef, right?